### PR TITLE
Drop assertion-time attestation.

### PR DESCRIPTION
**Pull request by agl**
We don't believe that there's a use for it now. The cases that wish to plumb this sort of data back can do so via other means.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1997.html" title="Last updated on Jan 10, 2024, 12:44 PM UTC (a0cfb6c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1997/a83c764...a0cfb6c.html" title="Last updated on Jan 10, 2024, 12:44 PM UTC (a0cfb6c)">Diff</a>

### CODE REVIEW

**Comment by rlin1** [comment]
Does that imply that device-bound keys in the "supplemental keys" extension cannot be attested?

**Comment by agl** [comment]
> Does that imply that device-bound keys in the "supplemental keys" extension cannot be attested?

No, they certainly can be. This change just reduces the number of places that can have attestations at assertion time from three to two.

**Comment by arianvp** [comment]
Big 👍 for this change.   Attestation should be stored at creation time and this simplifies the spec.  for DBKs creation time can overlap with assertion time of the main credential so it's still during "creation".

File: index.bs [hunk 1]
```diff
@@ -5057,24 +4996,13 @@ The privacy, security and operational characteristics of [=attestation=] depend
     operating environment, and so on.
 
 The [=attestation type=] and [=attestation statement format=] is chosen by the [=authenticator=];
-[=[RPS]=] can only signal their preferences by setting the {{PublicKeyCredentialCreationOptions/attestation}} and {{PublicKeyCredentialCreationOptions/attestationFormats}} parameters, or those with the same names in {{PublicKeyCredentialRequestOptions}}.
+[=[RPS]=] can only signal their preferences by setting the {{PublicKeyCredentialCreationOptions/attestation}} and {{PublicKeyCredentialCreationOptions/attestationFormats}} parameters.
```
Comment (emlun) [inline index.bs:4999]: Above this, [near the beginning of the "Attestation" section](https://github.com/w3c/webauthn/pull/1997/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985L5013), remains some language about "attestation is returned when a credential is created, and optionally exercised". See the diff that introduced it: https://github.com/w3c/webauthn/pull/1663/commits/20dd35c00a4caf07b1514c5495afaaf1482e04ae#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985L4308-R4358

File: index.bs [hunk 2]
```diff
@@ -2710,23 +2676,19 @@ The steps for [=issuing a credential request to an authenticator=] are as follow
...
-                            |enterpriseAttestationPossible|, |attestationFormats|, and |authenticatorExtensions| as parameters.
+                            and |authenticatorExtensions| as parameters.
                     </dl>
 
             :   [=list/is empty=]
             ::  Using local configuration knowledge of the appropriate transport to use with |authenticator|, invoke the
                 [=authenticatorGetAssertion=] operation on |authenticator| with |rpId|, |clientDataHash|,
                 |userVerification|,
-                |enterpriseAttestationPossible|,
-                |attestationFormats|,
```
Comment (emlun) [inline index.bs:2729]: These need to be deleted from the [authenticatorGetAssertion parameters](https://github.com/w3c/webauthn/pull/1997/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985R4736-R4739) too.

File: index.bs (hunk 1 above)
Comment (emlun) [reply index.bs:4999]: Similarly, I think we can revert the change to the `<figcaption>` in that section too: https://github.com/w3c/webauthn/pull/1663/commits/20dd35c00a4caf07b1514c5495afaaf1482e04ae#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985L4319-R4368

**Review by emlun — CHANGES_REQUESTED** [review]

**Review by MasterKale — APPROVED** [review]

File: index.bs [hunk 3]
```diff
@@ -3026,7 +2986,6 @@ optionally evidence of [=user consent=] to a specific transaction.
         [SameObject] readonly attribute ArrayBuffer      authenticatorData;
         [SameObject] readonly attribute ArrayBuffer      signature;
         [SameObject] readonly attribute ArrayBuffer?     userHandle;
-        [SameObject] readonly attribute ArrayBuffer?     attestationObject;
```
Comment (MasterKale) [inline index.bs:3029]: This will need to be dropped from `AuthenticatorAssertionResponseJSON` as well (it was added there too back in #1920)

Comment (MasterKale) [inline index.bs:3029]: This will need to be dropped from `AuthenticatorAssertionResponseJSON` as well (it was added there too back in #1920)

**Review by MasterKale — CHANGES_REQUESTED** [review]

File: index.bs (hunk 1 above)
Comment (agl) [reply index.bs:4999]: Thanks. Done.

File: index.bs (hunk 2 above)
Comment (agl) [reply index.bs:2729]: Yep, and more needs to be tidied up from there too. Done.

File: index.bs (hunk 3 above)
Comment (agl) [reply index.bs:3029]: Done.

**Review by MasterKale — APPROVED** [review]
Looks good to me.

**Review by emlun — APPROVED** [review]
Heh, I spotted this looking strange in the HTML diff:

screenshot-2024-01-10T13:37:43+01:00

This was added in [meta-PR #1818](https://github.com/w3c/webauthn/pull/1812/files#diff-5e793325cd2bfc452e268a4aa2f02b4024dd9584bd1db3c2595f61f1ecf7b985R5508-R5512), and apparently Bikeshed now started pulling in `{{AuthenticatorAssertionResponse/attestationObject}}` as an external ref from the spec database since the member definition no longer exists within the document.

I've pushed commit a0cfb6c897df6fc85c939062c9d46ecb7ac8c302 which fixes this.

**Pull request merged**